python/thunder/rdds/series.py

```diff
@@ -56,6 +56,22 @@ def index(self):
         if self._index is None:
             self.populateParamsFromFirstRecord()
         return self._index
+        
+    @index.setter
+    def index(self, value):
+        try:
+            value[0]
+        except:
+            raise ValueError("Index must be a non-empty indexable object")
+        if self._index is None:
+            self.populateParametersFromFirstRecord()
+        try:
+            lenMatch = len(value) == len(self._index)
+        except:
+            raise ValueError("Index must be an object with a length")
```

Comment:
Using the existing classes is just fine if they accurately describe the error that you're seeing. I'm probably guilty of overusing `ValueError` myself (but "inappropriate value" describes, like, everything! https://docs.python.org/2/library/exceptions.html)

The one custom exception I can think of offhand in Thunder is a `FileNotFoundException` defined in reader.py. It's a little silly that this doesn't already exist I think - and depending on the exact context, having a missing file could result in either `IOError` or `OSError`. Personally, I haven't run into that many cases where I thought having an exception class of my very own would be all that useful. But don't let this stop you. :)